src/main/java/com/artipie/asto/google/GoogleStorage.java

```diff
@@ -107,15 +146,16 @@ public GoogleStorage(final Vertx vertx, final WebClient client, final String buc
     public CompletableFuture<Content> value(final Key key) {
         final ReactiveWriteStream<Buffer> stream = ReactiveWriteStream
             .writeStream(this.vertx.getDelegate());
-        this.client.get(String.format(GoogleStorage.GET_URL, this.bucket, key.string()))
+        this.client.get(
+            this.port,
+            this.host,
+            String.format(GoogleStorage.GET_REQUEST, this.bucket, key.string())
+        )
             .as(BodyCodec.pipe(WriteStream.newInstance(stream)))
             .rxSend().subscribe();
-        return  Flowable.fromPublisher(stream).flatMapCompletable(
-            buffer -> Completable.fromSingle(
-                Single.just(new Content.From(buffer.getBytes()))
-            ))
-            .to(CompletableInterop.await())
-            .thenApply(content -> (Content) content).toCompletableFuture();
```

Comment:
@Guseyn the subscription will be lost here: if client code requests cancellation of `CompletableFuture`, the underlying stream subscription won't be cancelled. `Single.subscribe()` returns `Disposable` susbcription, which can be canceled using `subscription.dispose()`

Reply:
@g4s8 This code is removed, instead I use:
```
final ReactiveWriteStream<Buffer> stream = ReactiveWriteStream
            .writeStream(this.vertx.getDelegate());
        this.client.get(
            this.port,
            this.host,
            String.format(GoogleStorage.GET_REQUEST, this.bucket, key.string())
        )
            .as(BodyCodec.pipe(WriteStream.newInstance(stream)))
            .rxSend().subscribe();
        final Flowable<ByteBuffer> flow = Flowable.fromPublisher(stream)
            .map(buffer -> ByteBuffer.wrap(buffer.getBytes()));
```
and then I return the response:
```
return CompletableFuture.supplyAsync(() -> new Content.From(flow))
```

Reply:
@g4s8 check the replies

Reply:
@Guseyn I mean the subscription won't be canceled on `CompletableFuture` cancelation, it's a bug